Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -14936,11 +14935,45 @@ void WebPageProxy::didFindTextManipulationItems(const Vector<WebCore::TextManipu
 
 void WebPageProxy::completeTextManipulation(const Vector<TextManipulationItem>& items, Function<void(bool allFailed, const Vector<TextManipulationControllerManipulationFailure>&)>&& completionHandler)
 {
-    if (!hasRunningProcess()) {
-        completionHandler(true, { });
-        return;
-    }
-    sendWithAsyncReply(Messages::WebPage::CompleteTextManipulation(items), WTFMove(completionHandler));
+    class TextManipulationCallbackAggregator final : public ThreadSafeRefCounted<TextManipulationCallbackAggregator, WTF::DestructionThread::MainRunLoop> {
+    public:
+        using Callback = CompletionHandler<void(bool, const Vector<TextManipulationControllerManipulationFailure>&)>;
+        static Ref<TextManipulationCallbackAggregator> create(Callback&& callback)
+        {
+            return adoptRef(*new TextManipulationCallbackAggregator(WTFMove(callback)));
+        }
+        ~TextManipulationCallbackAggregator()
+        {
+            ASSERT(RunLoop::isMain());
+            m_callback(m_allFailed, m_failures);
+        }
+        void addResult(bool allFailed, const Vector<TextManipulationControllerManipulationFailure>& failures)
+        {
+            if (allFailed)
+                return;
+
+            if (m_allFailed)
+                m_allFailed = false;
+            m_failures.appendVector(failures);
```

Comment:
Hm... this doesn't seem right to me. If one of the process fails all of them, we'd get `allFailed` true but that doesn't necessarily mean `m_allFailed` is set to `true`. In that case, we need to generate errors for all the items which have thus far failed.

Reply:
Fixed by making web process also report the item indexes that have been successfully replaced, so that UI process can compute the state of each item and generate errors for the ones that are not successfully replaced (and don't have an explicit error).